common/src/main/java/feast/common/function/Store.java

```diff
@@ -0,0 +1,163 @@
+   * @param exclude flag to determine if subscriptions with exclusion flag should be returned
+   * @return List of Subscription class objects
+   */
+  public static List<StoreProto.Store.Subscription> parseSubscriptionFrom(
+      String subscriptions, boolean exclude) {
+    return Arrays.stream(subscriptions.split(","))
+        .map(subscriptionStr -> convertStringToSubscription(subscriptionStr, exclude))
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Accepts a Subscription class object and returns it in string format
+   *
+   * @param sub Subscription class to be converted to string format
+   * @return String formatted Subscription class
+   */
+  public static String parseSubscriptionFrom(StoreProto.Store.Subscription sub) {
+    if (sub.getName().isEmpty() || sub.getProject().isEmpty()) {
+      throw new IllegalArgumentException(
+          String.format("Missing arguments in subscription string: %s", sub.toString()));
+    }
+
+    return String.format("%s:%s:%s", sub.getProject(), sub.getName(), sub.getExclude());
+  }
+
+  /**
+   * Accepts a exclude parameter to determine whether to return subscriptions that are excluded.
+   *
+   * @param sub String formatted Subscription to be converted to Subscription class
+   * @param exclude flag to determine if subscriptions with exclusion flag should be returned
+   * @return Subscription class with its respective attributes
+   */
+  private static StoreProto.Store.Subscription convertStringToSubscription(
+      String sub, boolean exclude) {
+    if (sub.equals("")) {
+      return StoreProto.Store.Subscription.newBuilder().build();
+    }
+    String[] split = sub.split(":");
+    if (split.length == 2) {
+      // Backward compatibility check
+      return StoreProto.Store.Subscription.newBuilder()
+          .setProject(split[0])
+          .setName(split[1])
+          .setExclude(false)
+          .build();
+    }
+
+    if (split.length == 3) {
+      // If exclusion flag is set to true
+      if (exclude && Boolean.parseBoolean(split[2])) {
+        return StoreProto.Store.Subscription.newBuilder().build();
+      }
+    }
+    return StoreProto.Store.Subscription.newBuilder()
+        .setProject(split[0])
+        .setName(split[1])
+        .setExclude(Boolean.parseBoolean(split[2]))
+        .build();
+  }
+
+  /**
+   * The current use of this function is to determine whether a FeatureRow is subscribed to a
+   * Featureset.
+   *
+   * @param subscriptions List of Subscriptions available in Store
+   * @param projectName Project name used for matching Subscription's Project
+   * @param featureSetName Featureset name used for matching Subscription's Featureset
+   * @return boolean flag to signify if FeatureRow is subscribed to Featureset
+   */
+  public static boolean isSubscribedToFeatureSet(
+      List<StoreProto.Store.Subscription> subscriptions,
+      String projectName,
+      String featureSetName) {
```

Comment:
Isn't a feature reference as an argument more natural here?

Reply:
So, for this function, its written to abstract out the usage in these 2 places, [1](https://github.com/feast-dev/feast/blob/master/ingestion/src/main/java/feast/ingestion/transform/FeatureRowToStoreAllocator.java#L78) and [2](https://github.com/feast-dev/feast/blob/master/ingestion/src/main/java/feast/ingestion/transform/specs/FilterRelevantFunction.java#L46). Wanted to keep the original intent of the function, unless I try to refactor the parts which are using this as well.